pkg/health/health.go

```diff
@@ -28,24 +30,33 @@ type Result struct {
 	Output  string
 }
 
-func (h ConsulHealthChecker) Service(serviceID string) (map[string][]Result, error) {
-	entries, _, err := h.client.Health().Service(serviceID, "", false, nil)
+func (h ConsulHealthChecker) Service(serviceID string) (map[string][][]Result, error) {
+	catalogEntries, _, err := h.client.Health().Service(serviceID, "", false, nil)
 	if err != nil {
 		return nil, util.Errorf("/health/service failed for %q: %s", serviceID, err)
 	}
 
-	ret := make(map[string][]Result)
-	for _, entry := range entries {
-		ret[entry.Node.Node] = make([]Result, 0, len(entry.Checks))
+	ret := make(map[string][][]Result)
+	for _, entry := range catalogEntries {
+		// get entry in KV store for this service/node
+		kvEntry, err := h.consulStore.GetHealth(serviceID, entry.Node.Node)
```

Comment:
This results in N queries even though we know we can query for the health of the service using one. Worth refactoring.

Also, this implementation implies that all services will have an entry in the catalog but this is definitely not the case.